persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -274,7 +274,9 @@ public static RuntimeException toExternal(org.apache.cassandra.exceptions.Cassan
       case IS_BOOTSTRAPPING:
         return new IsBootstrappingException();
       case TRUNCATE_ERROR:
-        return new TruncateException(e.getCause());
+        return e.getCause() == null
+            ? new TruncateException(e.getMessage())
+            : new TruncateException(e.getCause());
```

Comment:
It looks like one of these cases will add the `Error during truncate: ` prefix to the message, but the other will not. I suppose it would be nicer to keep the final messages consistent for both of these cases.

Reply:
Also, do we want to extract just the first level cause or the root cause exception?

Reply:
And another point: since we're removing some exception information, what do you think about adding the original exception to `Throwable.addSuppressed(...)` on the `TruncateException`? (probably applies to other cases too).

Reply:
> It looks like one of these cases will add the Error during truncate: prefix to the message, but the other will not. I suppose it would be nicer to keep the final messages consistent for both of these cases.